### PR TITLE
added hindi translation for js environment requirement

### PR DESCRIPTION
**Pull request by AlokTakshak**

### CODE REVIEW

File: content/blog/2019-08-15-new-react-devtools.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ It also offers full support for React Hooks, including inspecting nested objects
 * `0`-`0.61`: Not supported
 * `0.62`: Will be supported (when 0.62 is released)
 
-## How do I get the new DevTools?
+## How do I get the new DevTools? {#how-do-i-get-the-new-devtools}
```
[the remark this answers is not in the document]
Comment (AlokTakshak) [reply content/blog/2019-08-15-new-react-devtools.md]: > `{#how-do-i-get-the-new-devtools}` remove this.
> 
> Remove all the link titles, this should go to main repo and it will be merged from there.

this actually  came from running yarn check-all , i ran it because i thought it would check all necessary rule to be followed

**Comment by AlokTakshak** [comment]
> Added some review.
> Also you need to translate complete file in hindi.
> 
> Title links should go to the parent repository https://github.com/reactjs/reactjs.org

can you tell me what i need to convert in hindi , and which title link should go to parent repo. I didn't get what needs to be modified

**Comment by arshadkazmi42** [comment]
> > Added some review.
> > Also you need to translate complete file in hindi.
> > Title links should go to the parent repository https://github.com/reactjs/reactjs.org
> 
> can you tell me what i need to convert in hindi , and which title link should go to parent repo. I didn't get what needs to be modified

All the text in the file should be translated to hindi. You can check this PR for reference
https://github.com/reactjs/hi.reactjs.org/pull/41/files

**Comment by AlokTakshak** [comment]
> > > Added some review.
> > > Also you need to translate complete file in hindi.
> > > Title links should go to the parent repository https://github.com/reactjs/reactjs.org
> > 
> > 
> > can you tell me what i need to convert in hindi , and which title link should go to parent repo. I didn't get what needs to be modified
> 
> All the text in the file should be translated to hindi. You can check this PR for reference
> https://github.com/reactjs/hi.reactjs.org/pull/41/files

English Words i have used are React, Map, Set, requestAnimationFrame, core-js, babel-pollyfill, pollyfill
The PR you mentioned is also using React, UI, React DOM , setInterval, ReactDOM.render(), Text, callback etc.
if you want i can change  core-js, babel-pollyfill, pollyfill but rest are React and Javascript Specific.Please tell me what i need to change because i am not able to figure out of these word which i should change.

File: content/blog/2019-08-15-new-react-devtools.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ author: [bvaughn]
 ---
 We are excited to announce a new release of the React Developer Tools, available today in Chrome, Firefox, and (Chromium) Edge!
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: Translate this to hindi

Ex: `हम Chrome, Firefox और (Chromium) Edge में आज उपलब्ध React डेवलपर टूल्स की एक नई रिलीज़ की घोषणा करने के लिए उत्साहित हैं!`

Similar to this, complete file needs to be translated.

File: content/blog/2019-08-15-new-react-devtools.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ author: [bvaughn]
...
 
-## What's changed?
+## What's changed? {#whats-changed}
 
 A lot has changed in version 4!
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: Translate this to hindi

File: content/blog/2019-08-15-new-react-devtools.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ author: [bvaughn]
...
 At a high level, this new version should offer significant performance gains and an improved navigation experience.
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: Translate this to hindi

File: content/blog/2019-08-15-new-react-devtools.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ It also offers full support for React Hooks, including inspecting nested objects
 
 [Visit the interactive tutorial](https://react-devtools-tutorial.now.sh/) to try out the new version or [see the changelog](https://github.com/facebook/react/blob/master/packages/react-devtools/CHANGELOG.md#400-august-15-2019) for demo videos and more details.
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: `Visit the interactive tutorial` translate this to hindi
`see the changelog` translate this to hindi

File: content/blog/2019-08-15-new-react-devtools.md [hunk 6]
```diff
@@ -14,7 +14,7 @@ It also offers full support for React Hooks, including inspecting nested objects
...
 
-## Which versions of React are supported?
+## Which versions of React are supported? {#which-versions-of-react-are-supported}
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: Translate the title to hindi

Comment (AlokTakshak) [reply content/blog/2019-08-15-new-react-devtools.md]: Currently my task was to translate only javascript-environment  file, these files only got effect because i ran yarn-check all, my task was to only change https://github.com/reactjs/hi.reactjs.org/blob/master/content/docs/reference-javascript-environment-requirements.md

Comment (arshadkazmi42) [reply content/blog/2019-08-15-new-react-devtools.md]: > Currently my task was to translate only javascript-environment file, these files only got effect because i ran yarn-check all, my task was to only change https://github.com/reactjs/hi.reactjs.org/blob/master/content/docs/reference-javascript-environment-requirements.md

Oh. Alright. I thought you are working on all those pages. Let me start with the review then. Sorry for the confusion

File: content/blog/2019-08-15-new-react-devtools.md [hunk 7]
```diff
@@ -4,7 +4,7 @@ author: [bvaughn]
 ---
 We are excited to announce a new release of the React Developer Tools, available today in Chrome, Firefox, and (Chromium) Edge!
 
-## What's changed?
+## What's changed? {#whats-changed}
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: `## What's changed? {#whats-changed}` => `## What's changed?`

since you are not working on this page. let's revert this change

File: content/blog/2019-08-15-new-react-devtools.md (hunk 6 above)
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: Same as above

`## Which versions of React are supported? {#which-versions-of-react-are-supported}` => `## Which versions of React are supported?`

File: content/blog/2019-08-15-new-react-devtools.md (hunk 1 above)
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: same as above

`## How do I get the new DevTools? {#how-do-i-get-the-new-devtools}` => `## How do I get the new DevTools?`

File: content/blog/2019-08-15-new-react-devtools.md [hunk 8]
```diff
@@ -37,14 +37,14 @@ If you use the standalone shell (e.g. in React Native or Safari), you can instal
 npm install -g react-devtools@^4
 ```
 
-## Where did all of the DOM elements go?
+## Where did all of the DOM elements go? {#where-did-all-of-the-dom-elements-go}
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: same as above

`## Where did all of the DOM elements go? {#where-did-all-of-the-dom-elements-go}` => `## Where did all of the DOM elements go?`

File: content/blog/2019-08-15-new-react-devtools.md [hunk 9]
```diff
@@ -37,14 +37,14 @@ If you use the standalone shell (e.g. in React Native or Safari), you can instal
...
 
 The new DevTools provides a way to filter components from the tree to make it easier to navigate deeply nested hierarchies.
 Host nodes (e.g. HTML `<div>`, React Native `<View>`) are *hidden by default*, but this filter can be disabled:
 
 ![DevTools component filters](../images/blog/devtools-component-filters.gif)
 
-## How do I get the old version back?
+## How do I get the old version back? {#how-do-i-get-the-old-version-back}
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: `## How do I get the old version back? {#how-do-i-get-the-old-version-back}` => `## How do I get the old version back?`

File: content/blog/2019-08-15-new-react-devtools.md [hunk 10]
```diff
@@ -70,7 +70,7 @@ yarn build:extension
 # Follow the on-screen instructions to complete installation
 ```
 
-## Thank you!
+## Thank you! {#thank-you}
```
Comment (arshadkazmi42) [inline content/blog/2019-08-15-new-react-devtools.md]: `## Thank you! {#thank-you}` => `## Thank you!`

File: content/docs/reference-javascript-environment-requirements.md [hunk 11]
```diff
@@ -1,14 +1,14 @@
 ---
 id: javascript-environment-requirements
-title: JavaScript Environment Requirements
+title: जावास्क्रिप्ट पर्यावरण आवश्यकताएँ
```
Comment (arshadkazmi42) [inline content/docs/reference-javascript-environment-requirements.md]: `पर्यावरण` changes the contextual meaning of environment. let keep it

`पर्यावरण` => `एनवायरनमेंट`

File: content/docs/reference-javascript-environment-requirements.md [hunk 12]
```diff
@@ -1,14 +1,14 @@
...
 ---
 
-React 16 depends on the collection types [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) and [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set). If you support older browsers and devices which may not yet provide these natively (e.g. IE < 11) or which have non-compliant implementations (e.g. IE 11), consider including a global polyfill in your bundled application, such as [core-js](https://github.com/zloirock/core-js) or [babel-polyfill](https://babeljs.io/docs/usage/polyfill/).
+React 16 संग्रह के प्रकार पर निर्भर करता है [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) और [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)। यदि आप पुराने ब्राउज़रों और उपकरणों का समर्थन करते हैं, जो अभी तक ये मूल रूप से प्रदान नहीं कर सकते हैं (जैसे IE <11) या जिनके पास गैर-अनुरूप कार्यान्वयन हैं (उदाहरण IE 11), तो अपने बंडल किए गए एप्लिकेशन में एक वैश्विक polyfill शामिल करें, जैसे कि [core-js](https://github.com/zloirock/core-js) या [babel-polyfill](https://babeljs.io/docs/usage/polyfill/)।
```
Comment (arshadkazmi42) [inline content/docs/reference-javascript-environment-requirements.md]:
```
React 16 संग्रह के प्रकार पर निर्भर करता है [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) और [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)।
```
=>
```
React 16 [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) और [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) संग्रह के प्रकार पर निर्भर करता है ।
```

How about this?

File: content/docs/reference-javascript-environment-requirements.md [hunk 13]
```diff
@@ -1,14 +1,14 @@
...
 
-A polyfilled environment for React 16 using core-js to support older browsers might look like:
+पुराने ब्राउज़र का समर्थन करने के लिए core-js का उपयोग करके React 16 के लिए एक polyfilled वातावरण ऐसा दिख सकता है:
```
Comment (arshadkazmi42) [inline content/docs/reference-javascript-environment-requirements.md]: `वातावरण` => `एनवायरनमेंट`

File: content/docs/reference-javascript-environment-requirements.md [hunk 14]
```diff
@@ -23,8 +23,8 @@ ReactDOM.render(
 );
 ```
 
-React also depends on `requestAnimationFrame` (even in test environments).  
-You can use the [raf](https://www.npmjs.com/package/raf) package to shim `requestAnimationFrame`:
+React `requestAnimationFrame` पर भी निर्भर करती है (परीक्षण वातावरण में भी)।
+आप `requestAnimationFrame` के समर्थन के लिए [raf](https://www.npmjs.com/package/raf) package का उपयोग कर सकते हैं :
```
Comment (arshadkazmi42) [inline content/docs/reference-javascript-environment-requirements.md]: `package` => `पैकेज`

Comment (arshadkazmi42) [inline content/docs/reference-javascript-environment-requirements.md]: `सकते हैं : ` => `सकते हैं :`

File: content/docs/reference-profiler.md [hunk 15]
```diff
@@ -16,7 +16,7 @@ Its purpose is to help identify parts of an application that are slow and may be
 > To opt into production profiling, React provides a special production build with profiling enabled.
 > Read more about how to use this build at [fb.me/react-profiling](https://fb.me/react-profiling)
 
-## Usage
+## Usage {#usage}
```
Comment (arshadkazmi42) [inline content/docs/reference-profiler.md]: Revert this as you are not working on this page
`## Usage {#usage}` => `## Usage`

File: content/docs/reference-profiler.md [hunk 16]
```diff
@@ -70,7 +70,7 @@ render(
 >
 > Although `Profiler` is a light-weight component, it should be used only when necessary; each use adds some CPU and memory overhead to an application.
 
-## `onRender` Callback
+## `onRender` Callback {#onrender-callback}
```
Comment (arshadkazmi42) [inline content/docs/reference-profiler.md]:
```
## `onRender` Callback {#onrender-callback}
```
=>
```
## `onRender` Callback
```

File: content/docs/reference-javascript-environment-requirements.md (hunk 14 above)
Comment (AlokTakshak) [reply content/docs/reference-javascript-environment-requirements.md]: in hindi glossary  its is written no translation is required for package

Comment (arshadkazmi42) [reply content/docs/reference-javascript-environment-requirements.md]: Oh. Yes. Totally forgot about it.
Thanks for cross checking

File: content/docs/reference-javascript-environment-requirements.md [hunk 17]
```diff
@@ -1,14 +1,14 @@
...
 ---
 
-React 16 depends on the collection types [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) and [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set). If you support older browsers and devices which may not yet provide these natively (e.g. IE < 11) or which have non-compliant implementations (e.g. IE 11), consider including a global polyfill in your bundled application, such as [core-js](https://github.com/zloirock/core-js) or [babel-polyfill](https://babeljs.io/docs/usage/polyfill/).
+React 16 [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) और [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) संग्रह के प्रकार पर निर्भर करता है। यदि आप पुराने ब्राउज़रों और उपकरणों का समर्थन करते हैं, जो अभी तक ये मूल रूप से प्रदान नहीं कर सकते हैं (जैसे IE <11) या जिनके पास गैर-अनुरूप कार्यान्वयन हैं (उदाहरण IE 11), तो अपने बंडल किए गए एप्लिकेशन में एक वैश्विक polyfill शामिल करें, जैसे कि [core-js](https://github.com/zloirock/core-js) या [babel-polyfill](https://babeljs.io/docs/usage/polyfill/)।
```
Comment (saranshkataria) [inline content/docs/reference-javascript-environment-requirements.md]: संग्रह के प्रकार पर निर्भर करता है could be better as collection type in hindi to make it more understandable

Comment (saranshkataria) [reply content/docs/reference-javascript-environment-requirements.md]: ब्राउज़रों -> browsers

Comment (saranshkataria) [reply content/docs/reference-javascript-environment-requirements.md]: जो अभी तक ये मूल रूप से प्रदान नहीं कर सकते हैं -> जो अभी तक ye type natively (in hindi) प्रदान नहीं karte

Comment (saranshkataria) [reply content/docs/reference-javascript-environment-requirements.md]: गैर-अनुरूप -> too complex, could be non compliant in hindi too?

File: content/docs/reference-javascript-environment-requirements.md [hunk 18]
```diff
@@ -1,14 +1,14 @@
...
 
-A polyfilled environment for React 16 using core-js to support older browsers might look like:
+पुराने ब्राउज़र का समर्थन करने के लिए core-js का उपयोग करके React 16 के लिए एक polyfilled एनवायरनमेंट ऐसा दिख सकता है:
```
Comment (saranshkataria) [inline content/docs/reference-javascript-environment-requirements.md]: ऐसा दिख सकता है -> aisa dikhega

File: content/docs/reference-javascript-environment-requirements.md [hunk 19]
```diff
@@ -23,8 +23,8 @@ ReactDOM.render(
 );
 ```
 
-React also depends on `requestAnimationFrame` (even in test environments).  
-You can use the [raf](https://www.npmjs.com/package/raf) package to shim `requestAnimationFrame`:
+React `requestAnimationFrame` पर भी निर्भर करती है (परीक्षण एनवायरनमेंट में भी)।
+आप `requestAnimationFrame` के समर्थन के लिए [raf](https://www.npmjs.com/package/raf) package का उपयोग कर सकते हैं :
```
Comment (saranshkataria) [inline content/docs/reference-javascript-environment-requirements.md]: package could be converted to hindi package?

**Comment by saranshkataria** [comment]
Great work, added some comments and this is good to go once fixed!

File: content/docs/reference-javascript-environment-requirements.md (hunk 17 above)
Comment (AlokTakshak) [reply content/docs/reference-javascript-environment-requirements.md]: > ब्राउज़रों -> browsers

in Hindi Glosary we have a translated version of Browser
#2

Comment (AlokTakshak) [reply content/docs/reference-javascript-environment-requirements.md]: > ब्राउज़रों -> browsers

in Hindi Glosary we have a translated version of Browser
#2

File: content/docs/reference-javascript-environment-requirements.md (hunk 19 above)
Comment (AlokTakshak) [reply content/docs/reference-javascript-environment-requirements.md]: > package could be converted to hindi package?

in Hindi Glosary it is stated that  package should not be translated
#2

File: content/docs/reference-javascript-environment-requirements.md (hunk 17 above)
Comment (saranshkataria) [reply content/docs/reference-javascript-environment-requirements.md]: I mean ब्राउज़र्स, ब्राउज़रों is weird

**Comment by AlokTakshak** [comment]
> Great work, added some comments and this is good to go once fixed!

Done

**Comment by saranshkataria** [comment]
LGTM, Thank you!
